### PR TITLE
[Movie Details] Refactoring of movie description

### PR DESCRIPTION
**Pull request by Kachyr**
#166 task
The title was replaced with <Title/> component and the table with description was replaced on <Detail/> component.
Also, I have refactored some styles on MovieDetails.tsx and <Detail/> component.

### CODE REVIEW

File: src/components/Detail.tsx [hunk 1]
```diff
@@ -1,27 +1,20 @@
...
-      arrayOfContent: { name: string }[];
-    }
-  | {
-      title: string;
-      textContent: string | number;
-      arrayOfContent?: { name: string }[];
-    };
+type DetailProps = {
+  title: string;
+  textContent: string | number | undefined | null;
```
Comment (di-shu) [inline src/components/Detail.tsx:5]: if textContent takes 'undefined' and 'null', maybe this property is optional?

File: src/components/MovieDetails.tsx [hunk 2]
```diff
@@ -1,10 +1,12 @@
-import { useState, useEffect, ReactNode } from 'react';
+import { useState, useEffect } from 'react';
 
 import api from '../service/api';
 
 import ReadMore from './ReadMore';
 
 import '../styles/MovieDetails.scss';
+import Detail from './Detail';
```
Comment (di-shu) [inline src/components/MovieDetails.tsx:8]: You should order your imports.

File: src/styles/MovieDetails.scss [hunk 3]
```diff
@@ -8,41 +8,24 @@
...
     .movie-img {
       img {
-        width: 100%;
+        object-fit:fill;
         height: 100%;
         border-radius: 10px;
       }
     }
     .movie-details {
+
```
Comment (di-shu) [inline src/styles/MovieDetails.scss:19]: remove empty line

**Review by di-shu — CHANGES_REQUESTED** [review]

File: src/components/Detail.tsx (hunk 1 above)
Comment (Kachyr) [reply src/components/Detail.tsx:5]: I'm monitoring this property inside component

File: src/components/MovieDetails.tsx (hunk 2 above)
Comment (Kachyr) [reply src/components/MovieDetails.tsx:8]: done

File: src/styles/MovieDetails.scss (hunk 3 above)
Comment (Kachyr) [reply src/styles/MovieDetails.scss]: done

File: src/components/Detail.tsx (hunk 1 above)
Comment (oleksandrstarov) [reply src/components/Detail.tsx:5]: textContent: string | number | undefined | null; >> textContent?: string | number | null;
It is easier to see that property is optional

File: src/components/MovieDetails.tsx [hunk 4]
```diff
@@ -94,19 +81,20 @@ const MovieDetails = ({ match }: MovieDetailsProps) => {
...
-              {renderMovieInfo('Popularity', popularity)}
-            </table>
+            <Detail title="Original title" textContent={original_title} />
+            <Detail title="Tagline" textContent={tagline} />
+            <Detail title="release_date" textContent={release_date} />
+            <Detail title="Status" textContent={status} />
+            <Detail title="Budget" textContent={budget} />
+            <Detail
+              title="Country"
+              textContent={production_countries[0].name}
```
[the remark this answers is not in the document]
Comment (Kachyr) [reply src/components/MovieDetails.tsx]: That's right, thank you

File: src/components/Detail.tsx (hunk 1 above)
Comment (solomiia-sahala) [reply src/components/Detail.tsx:5]: i would recommend you to change textContent variable to some other name, "textContent" suits better for String type.

**Review by solomiia-sahala — CHANGES_REQUESTED** [review]

**Pull request merged**